cli/cross-chain.ts

```diff
@@ -0,0 +1,63 @@
+import { L1ToL2MessageGasEstimator } from '@arbitrum/sdk'
+import { BigNumber, providers } from 'ethers'
+import { parseEther } from 'ethers/lib/utils'
+
+import { logger } from './logging'
+
+export const l1ToL2ChainIdMap = {
+  '1': '42161',
+  '4': '421611',
+  '5': '421613',
+  '1337': '412346',
+}
+
+export const l2ChainIds = Object.values(l1ToL2ChainIdMap).map(Number)
+export const l2ToL1ChainIdMap = Object.fromEntries(
+  Object.entries(l1ToL2ChainIdMap).map(([k, v]) => [v, k]),
+)
+
+export const chainIdIsL2 = (chainId: number | string): boolean => {
+  return l2ChainIds.includes(Number(chainId))
+}
+
+interface L2GasParams {
+  maxGas: BigNumber
+  gasPriceBid: BigNumber
+  maxSubmissionCost: BigNumber
+}
+
+export const estimateRetryableTxGas = async (
+  l1Provider: providers.Provider,
+  l2Provider: providers.Provider,
+  gatewayAddress: string,
+  l2Dest: string,
+  depositCalldata: string,
+  opts: L2GasParams,
+): Promise<L2GasParams> => {
+  const autoEstimate = opts && (!opts.maxGas || !opts.gasPriceBid || !opts.maxSubmissionCost)
```

Comment:
Is the `opts &&` needed here? I think typescript won't allow it to be null anyways (unless you mark it as optional in the function definition), but in any case if opts is nullish we would want to auto-estimate, wouldn't we?